rasa/core/processor.py

```diff
@@ -180,6 +180,8 @@ async def run_action_extract_slots(
             f"Default action '{ACTION_EXTRACT_SLOTS}' was executed, "
             f"resulting in {len(extraction_events)} events: {events_as_str}"
         )
+        await self.execute_side_effects(extraction_events, tracker, output_channel)
```

Comment:
I see that the other side effects are scheduling / cancelling reminders - do you know if the behaviour for either of those now changes?
Is it possible to include unit tests for the expected behaviour of those two, once we know what it is?

Reply:
Good question, I should only apply the first side-effect (sending bot messages), since the other events are disallowed during validation (only `SlotSet` and `BotUttered` are allowed event types to be returned during slot validation, see [here](await self._send_bot_messages(events, tracker, output_channel))).

Reply:
Gotcha, that makes sense. 
One thing I am confused by, why do users want to return `BotUttered` events instead of returning `responses` directly? Or is this different for slot validation than for other custom actions?

Reply:
> why do users want to return BotUttered events instead of returning responses directly?

You mean why wouldn't they use `dispatcher.utter_message` instead of returning the `BotUttered` event? Not sure 😄  It seems both approaches are valid, so we wanted to enable that.